(module name): short issue description

**Describe the bug**
I have a device configured AWS IoT Greengrass V2 with the PKCS11 Provider over a TPM 2 w/ tpm2-tools.
The device with such set-up has passed the AWS IoT Greengrass V2 device qualification tests.
The device connects correctly to AWS IoT. When I try a module deploy action to the device, the following Java exception is reported.

2022-02-16T05:38:30.997Z [INFO] (pool-2-thread-7) com.aws.greengrass.mqttclient.AwsIotMqttClient: Connecting to AWS IoT Core. {clientId=REGATE-10-14-GreengrassCoreHSM}
2022-02-16T05:38:41.108Z [INFO] (Thread-4) com.aws.greengrass.mqttclient.AwsIotMqttClient: Connecting to AWS IoT Core. {clientId=REGATE-10-14-GreengrassCoreHSM}
2022-02-16T05:38:41.109Z [INFO] (Thread-4) com.aws.greengrass.mqttclient.AwsIotMqttClient: Connection interrupted. {clientId=REGATE-10-14-GreengrassCoreHSM}
2022-02-16T05:38:42.072Z [INFO] (Thread-4) com.aws.greengrass.mqttclient.AwsIotMqttClient: Successfully connected to AWS IoT Core. {clientId=REGATE-10-14-GreengrassCoreHSM, sessionPresent=false}
2022-02-16T05:38:43.150Z [INFO] (Thread-4) com.aws.greengrass.deployment.IotJobsHelper: Job status update was accepted. {Status=IN_PROGRESS, ThingName=REGATE-10-14-GreengrassCoreHSM, JobId=bb4eaa53-dd9f-43ea-b84d-425bb71ac880}
2022-02-16T05:38:43.409Z [INFO] (Thread-4) com.aws.greengrass.deployment.IotJobsHelper: Received Iot job description. {Status=IN_PROGRESS, queueAt=Tue Feb 15 21:00:39 UTC 2022, ThingName=REGATE-10-14-GreengrassCoreHSM, JobId=bb4eaa53-dd9f-43ea-b84d-425bb71ac880}
2022-02-16T05:38:43.411Z [INFO] (Thread-4) com.aws.greengrass.deployment.IotJobsHelper: Duplicate or outdated job notification. Ignoring.. {ThingName=REGATE-10-14-GreengrassCoreHSM, JobId=bb4eaa53-dd9f-43ea-b84d-425bb71ac880}
2022-02-16T05:38:43.830Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentService: Started deployment execution. {serviceName=DeploymentService, currentState=RUNNING, deployment=bb4eaa53-dd9f-43ea-b84d-425bb71ac880}
2022-02-16T05:38:43.899Z [INFO] (pool-2-thread-7) com.aws.greengrass.status.FleetStatusService: fss-status-update-published. Status update published to FSS. {serviceName=FleetStatusService, currentState=RUNNING}
2022-02-16T05:38:58.832Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentService: Current deployment finished. {DeploymentId=bb4eaa53-dd9f-43ea-b84d-425bb71ac880, serviceName=DeploymentService, currentState=RUNNING}
2022-02-16T05:38:58.942Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentStatusKeeper: Stored deployment status. {DeploymentId=bb4eaa53-dd9f-43ea-b84d-425bb71ac880, DeploymentStatus=SUCCEEDED}
2022-02-16T05:38:58.944Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.IotJobsHelper: Updating status of persisted deployment. {Status=SUCCEEDED, StatusDetails={detailed-deployment-status=SUCCESSFUL}, ThingName=REGATE-10-14-GreengrassCoreHSM, JobId=bb4eaa53-dd9f-43ea-b84d-425bb71ac880}
2022-02-16T05:38:59.386Z [INFO] (Thread-4) com.aws.greengrass.deployment.IotJobsHelper: Job status update was accepted. {Status=SUCCEEDED, ThingName=REGATE-10-14-GreengrassCoreHSM, JobId=bb4eaa53-dd9f-43ea-b84d-425bb71ac880}
2022-02-16T05:38:59.696Z [INFO] (pool-2-thread-15) com.aws.greengrass.status.FleetStatusService: fss-status-update-published. Status update published to FSS. {serviceName=FleetStatusService, currentState=RUNNING}
2022-02-16T05:38:59.700Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentDirectoryManager: Persist link to last deployment. {link=/opt/aws/greengrass/v2/deployments/previous-success}
2022-02-16T05:38:59.708Z [INFO] (Thread-4) com.aws.greengrass.deployment.IotJobsHelper: Received empty jobs in notification . {ThingName=REGATE-10-14-GreengrassCoreHSM}
2022-02-16T05:50:17.192Z [INFO] (Thread-4) com.aws.greengrass.deployment.ShadowDeploymentListener: Added shadow deployment job. {ID=arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1}
2022-02-16T05:50:29.707Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentService: Received deployment in the queue. {deploymentId=arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1, DeploymentType=SHADOW, serviceName=DeploymentService, currentState=RUNNING}
2022-02-16T05:50:29.708Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentService: Received deployment document in queue. {document={"deploymentId":"04c73eec-28e5-46b5-abda-e9eeec062476","schemaDate":"2021-05-17","configurationArn":"arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1","creationTimestamp":1644990617220,"components":{"aws.greengrass.Cli":{"version":"2.5.3"}},"failureHandlingPolicy":"ROLLBACK","requiredCapabilities":[],"componentUpdatePolicy":{"timeout":60,"action":"NOTIFY_COMPONENTS"},"configurationValidationPolicy":{"timeout":60}}, serviceName=DeploymentService, currentState=RUNNING}
2022-02-16T05:50:29.750Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentStatusKeeper: Stored deployment status. {DeploymentId=arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1, DeploymentStatus=IN_PROGRESS}
2022-02-16T05:50:29.756Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.ShadowDeploymentListener: Updated reported state for deployment. {CONFIGURATION_ARN=arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1, status=IN_PROGRESS}
2022-02-16T05:50:29.759Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentDirectoryManager: Create work directory for new deployment. {deploymentId=04c73eec-28e5-46b5-abda-e9eeec062476, link=/opt/aws/greengrass/v2/deployments/ongoing, directory=/opt/aws/greengrass/v2/deployments/04c73eec-28e5-46b5-abda-e9eeec062476}
2022-02-16T05:50:29.762Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentDirectoryManager: Persist deployment metadata. {file=/opt/aws/greengrass/v2/deployments/04c73eec-28e5-46b5-abda-e9eeec062476/deployment_metadata.json, deploymentId=04c73eec-28e5-46b5-abda-e9eeec062476}
2022-02-16T05:50:29.800Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentService: Started deployment execution. {serviceName=DeploymentService, currentState=RUNNING, deployment=arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1}
2022-02-16T05:50:29.800Z [INFO] (pool-2-thread-19) com.aws.greengrass.deployment.DeploymentService: deployment-task-execution. Starting deployment task. {Deployment service config={ComponentToGroups={aws.greengrass.Cli={arn:aws:greengrass:us-east-1:332696098873:configuration:thinggroup/EurotechGreengrassCoreGroup:4=thinggroup/EurotechGreengrassCoreGroup}, aws.greengrass.crypto.Pkcs11Provider={arn:aws:greengrass:us-east-1:332696098873:configuration:thinggroup/EurotechGreengrassCoreGroup:4=thinggroup/EurotechGreengrassCoreGroup}, aws.greengrass.Nucleus={arn:aws:greengrass:us-east-1:332696098873:configuration:thinggroup/EurotechGreengrassCoreGroup:4=thinggroup/EurotechGreengrassCoreGroup}}, dependencies=[], GroupToRootComponents={thinggroup/EurotechGreengrassCoreGroup={aws.greengrass.Cli={groupConfigArn=arn:aws:greengrass:us-east-1:332696098873:configuration:thinggroup/EurotechGreengrassCoreGroup:4, groupConfigName=thinggroup/EurotechGreengrassCoreGroup, version=2.5.3}, aws.greengrass.crypto.Pkcs11Provider={groupConfigArn=arn:aws:greengrass:us-east-1:332696098873:configuration:thinggroup/EurotechGreengrassCoreGroup:4, groupConfigName=thinggroup/EurotechGreengrassCoreGroup, version=2.0.0}}}, runtime={ProcessedDeployments={}}, version=0.0.0}, deploymentId=04c73eec-28e5-46b5-abda-e9eeec062476, serviceName=DeploymentService, currentState=RUNNING}
2022-02-16T05:50:32.984Z [INFO] (pool-2-thread-20) com.aws.greengrass.componentmanager.DependencyResolver: resolve-group-dependencies-start. Start to resolve group dependencies. {targetComponents=[aws.greengrass.Cli], componentToVersionRequirements={aws.greengrass.Cli={thing/REGATE-10-14-GreengrassCoreHSM==2.5.3, thinggroup/EurotechGreengrassCoreGroup==2.5.3}, aws.greengrass.crypto.Pkcs11Provider={thinggroup/EurotechGreengrassCoreGroup==2.0.0}, aws.greengrass.Nucleus={aws.greengrass.Cli=>=2.5.0 <2.6.0, aws.greengrass.crypto.Pkcs11Provider=>=2.5.3 <2.6.0}}}
2022-02-16T05:50:32.998Z [INFO] (pool-2-thread-20) com.aws.greengrass.componentmanager.ComponentManager: Found running component which meets the requirement and use it.. {ComponentIdentifier=aws.greengrass.Cli-v2.5.3}
2022-02-16T05:50:32.999Z [INFO] (pool-2-thread-20) com.aws.greengrass.componentmanager.ComponentManager: Found the best local candidate that satisfies the requirement.. {LocalCandidateId=aws.greengrass.Cli-v2.5.3}
2022-02-16T05:50:34.213Z [INFO] (pool-2-thread-20) com.aws.greengrass.componentmanager.ComponentManager: resolve-component-version-end. Resolved component version.. {ResolvedComponent=aws.greengrass.Cli-v2.5.3}
2022-02-16T05:50:34.319Z [INFO] (pool-2-thread-20) com.aws.greengrass.componentmanager.DependencyResolver: resolve-group-dependencies-finish. Finish resolving group dependencies. {resolvedComponents={aws.greengrass.Cli=ComponentMetadata(componentIdentifier=aws.greengrass.Cli-v2.5.3, dependencies={aws.greengrass.Nucleus=>=2.5.0 <2.6.0}), aws.greengrass.crypto.Pkcs11Provider=ComponentMetadata(componentIdentifier=aws.greengrass.crypto.Pkcs11Provider-v2.0.0, dependencies={aws.greengrass.Nucleus=>=2.5.3 <2.6.0}), aws.greengrass.Nucleus=ComponentMetadata(componentIdentifier=aws.greengrass.Nucleus-v2.5.3, dependencies={})}, componentToVersionRequirements={aws.greengrass.Cli={thing/REGATE-10-14-GreengrassCoreHSM==2.5.3, thinggroup/EurotechGreengrassCoreGroup==2.5.3}, aws.greengrass.crypto.Pkcs11Provider={thinggroup/EurotechGreengrassCoreGroup==2.0.0}, aws.greengrass.Nucleus={aws.greengrass.Cli=>=2.5.0 <2.6.0, aws.greengrass.crypto.Pkcs11Provider=>=2.5.3 <2.6.0}}}
2022-02-16T05:50:34.361Z [INFO] (pool-2-thread-20) com.aws.greengrass.componentmanager.ComponentManager: prepare-package-start. {packageIdentifier=aws.greengrass.Cli-v2.5.3}
2022-02-16T05:50:34.565Z [INFO] (pool-2-thread-20) com.aws.greengrass.componentmanager.ComponentManager: prepare-package-start. {packageIdentifier=aws.greengrass.crypto.Pkcs11Provider-v2.0.0}
2022-02-16T05:50:34.579Z [INFO] (pool-2-thread-20) com.aws.greengrass.componentmanager.ComponentManager: prepare-package-start. {packageIdentifier=aws.greengrass.Nucleus-v2.5.3}
2022-02-16T05:50:44.801Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentService: Current deployment finished. {DeploymentId=arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1, serviceName=DeploymentService, currentState=RUNNING}
2022-02-16T05:50:44.802Z [ERROR] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentService: Deployment task throws unknown exception. {DeploymentId=arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1, serviceName=DeploymentService, currentState=RUNNING}
java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.String
        at com.aws.greengrass.security.provider.pkcs11.PKCS11CryptoKeyService.isBootstrapRequired(PKCS11CryptoKeyService.java:443)
        at com.aws.greengrass.deployment.bootstrap.BootstrapManager.serviceBootstrapRequired(BootstrapManager.java:332)
        at com.aws.greengrass.deployment.bootstrap.BootstrapManager.lambda$isBootstrapRequired$0(BootstrapManager.java:131)
        at java.util.HashMap.forEach(HashMap.java:1290)
        at com.aws.greengrass.deployment.bootstrap.BootstrapManager.isBootstrapRequired(BootstrapManager.java:130)
        at com.aws.greengrass.deployment.activator.DeploymentActivatorFactory.getDeploymentActivator(DeploymentActivatorFactory.java:33)
        at com.aws.greengrass.deployment.DeploymentConfigMerger.mergeInNewConfig(DeploymentConfigMerger.java:80)
        at com.aws.greengrass.deployment.DefaultDeploymentTask.call(DefaultDeploymentTask.java:153)
        at com.aws.greengrass.deployment.DefaultDeploymentTask.call(DefaultDeploymentTask.java:48)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

2022-02-16T05:50:44.831Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentStatusKeeper: Stored deployment status. {DeploymentId=arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1, DeploymentStatus=FAILED}
2022-02-16T05:50:44.834Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.ShadowDeploymentListener: Updated reported state for deployment. {CONFIGURATION_ARN=arn:aws:greengrass:us-east-1:332696098873:configuration:thing/REGATE-10-14-GreengrassCoreHSM:1, status=FAILED}
2022-02-16T05:50:44.835Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentDirectoryManager: Persist link to last deployment. {link=/opt/aws/greengrass/v2/deployments/previous-failure}
2022-02-16T05:50:44.837Z [INFO] (pool-2-thread-15) com.aws.greengrass.deployment.DeploymentDirectoryManager: Clean up link to earlier deployment. {link=/opt/aws/greengrass/v2/deployments/previous-success}


**To Reproduce**
Configure a device with AWS IoT Greengrass V2 with the PKCS11 Provider over a TPM 2 w/ tpm2-tools.
Connect the device to AWS IoT Core.
From AWS IoT Console, start a Greengrass Module deployment.


**Environment**
- OS: Yocto
- JDK version: 1.8.0_312
- Nucleus version: 2.5.3
- PKCS#11 Provider version: 2.0.0

--- a/src/main/java/com/aws/greengrass/security/provider/pkcs11/PKCS11CryptoKeyService.java
+++ b/src/main/java/com/aws/greengrass/security/provider/pkcs11/PKCS11CryptoKeyService.java
@@ -439,8 +439,8 @@ public class PKCS11CryptoKeyService extends PluginService implements CryptoKeySp
             logger.atTrace().log("PKCS11 slot id changes, requires bootstrap");
             return true;
         }
-        char[] updatedUserPin = newConfiguration.get(USER_PIN_TOPIC) == null ? null
-                : ((String) newConfiguration.get(USER_PIN_TOPIC)).toCharArray();
+        char[] updatedUserPin = Coerce.toString(newConfiguration.get(USER_PIN_TOPIC)) == null ? null
+                : Coerce.toString(newConfiguration.get(USER_PIN_TOPIC)).toCharArray();
         if (!Arrays.equals(userPin, updatedUserPin)) {
             logger.atTrace().log("PKCS11 user pin changes, requires bootstrap");
             return true;

